src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -131,73 +102,40 @@ func (server *Server) Configure(newCfg *ServerConfig) {
 	server.cfg = newCfg
 }
 
-// CertContext encapsulates the certificates used by a Server
-type CertContext struct {
-	PKFile         string
-	ServerCertFile string
-	PK             *keyman.PrivateKey
-	ServerCert     *keyman.Certificate
-}
-
 func (server *Server) ListenAndServe() error {
-	err := server.CertContext.InitServerCert(strings.Split(server.Addr, ":")[0])
-	if err != nil {
-		return fmt.Errorf("Unable to init server cert: %s", err)
+	if server.Host != "" {
+		log.Debugf("Running as host %s", server.Host)
 	}
 
-	// Set up an enproxy Proxy
-	proxy := &enproxy.Proxy{
-		Dial: server.dialDestination,
-		Host: server.host,
+	fs := &fronted.Server{
+		Addr:                       server.Addr,
+		Host:                       server.Host,
+		ReadTimeout:                server.ReadTimeout,
+		WriteTimeout:               server.WriteTimeout,
+		CertContext:                server.CertContext,
+		AllowNonGlobalDestinations: server.AllowNonGlobalDestinations,
 	}
 
-	if server.host != "" {
-		log.Debugf("Running as host %s", server.host)
+	if server.cfg.Unencrypted {
+		log.Debug("Running in unencrypted mode")
+		fs.CertContext = nil
```

Comment:
To elaborate - we support unencrypted running on the server so that we can sit behind @aranhoide 's TLS terminator.  I made the client able to connect directly to such a server for testing purposes.